tutorials/developer_api_guide/my_trainable_tensor_subclass.py

```diff
@@ -0,0 +1,200 @@
+    """
+
+    @classmethod
+    def _quantize(
+        cls,
+        input_float: torch.Tensor,
+        layout_type: LayoutType,
+    ) -> MyDTypeLayout:
+        """
+        Convert from a floating point tensor (fp32/fp16/bf16) to the desired dtype.
+        """
+        mapping_type = MappingType.SYMMETRIC
+        block_size = input_float.shape
+        dtype = torch.int16
+        scale, _ = choose_qparams_affine(input_float, mapping_type, block_size, dtype)
+        int_data = (input_float / scale).to(torch.int8)
+        layout_tensor_ctr = cls.get_layout_tensor_constructor(type(layout_type))
+        return layout_tensor_ctr(int_data, scale, layout_type)
+
+    @classmethod
+    def from_float(
+        cls,
+        input_float: torch.Tensor,
+        layout_type: LayoutType = PlainLayoutType(),
+    ) -> "MyTrainableDTypeTensor":
+        """
+        Main entry point for creating a `MyTrainableDTypeTensor`.
+
+        This instantiates the tensor subclass in a differentiable constructor
+        to ensure gradients are passed to the tensor subclass properly during training.
+        """
+        return _ToMyTrainableDTypeTensor.apply(input_float, layout_type)
+
+class _ToMyTrainableDTypeTensor(torch.autograd.Function):
+    """ 
+    Differentiable constructor for `MyTrainableDTypeTensor`.
+    """
+
+    @staticmethod
+    def forward(
+        ctx: torch.autograd.function.FunctionCtx,
+        input_float: torch.Tensor,
+        layout_type: LayoutType,
+    ) -> "MyTrainableDTypeTensor":
+        layout_tensor = MyTrainableDTypeTensor._quantize(input_float, layout_type)
+        return MyTrainableDTypeTensor(
+            layout_tensor,
+            input_float.shape,
+            requires_grad=True,
+        )
+
+    @staticmethod
+    def backward(ctx, gy):
+        return gy, None
+
+to_my_trainable_dtype = MyTrainableDTypeTensor.from_float
+
+
+#####################################################
+# torch functional and aten operator implementation #
+#####################################################
+
+implements = MyTrainableDTypeTensor.implements
+
+class _QuantizedLinearOp(torch.autograd.Function):
+    """
+    Forward and backward definition for linear with quantized weights.
+    Weights are dequantized during both the forward and the backward passes.
+    """
+
+    @staticmethod
+    def forward(
+        ctx: torch.autograd.function.FunctionCtx,
+        input_tensor: torch.Tensor,
+        weight_tensor: torch.Tensor,
+    ) -> torch.Tensor:
+        assert isinstance(weight_tensor, MyTrainableDTypeTensor)
+        ctx.save_for_backward(input_tensor, weight_tensor)
+        weight_tensor = weight_tensor.dequantize()
+        return torch.nn.functional.linear(input_tensor, weight_tensor)
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        input_tensor, weight_tensor = ctx.saved_tensors
+        grad_input = torch.matmul(grad_output, weight_tensor.dequantize())
```

Comment:
sorry I guess I didn't meant to dequantize this, but was asking if this call into something like F.linear(grad_output, weight_tensor) and be dispatched to the quantized linear impl of weight_tensor.

Reply:
oh actually the old code was also doing dequantize math. I just rewrote it so it's cleaner. Not sure if I understand how we can dispatch to the quantized linear impl, since this is called from `F.linear(..., weight_subclass_tensor)` already?